dotty-scala-library/src/scala/StringContext.scala

```diff
@@ -0,0 +1,488 @@
+/*
```

Comment:
Why is this file needed? I tthought we already handled `macro ???` to allow compiling the scala 2 stdlb as is.

Reply:
We can compile them, but we assume that if they are going to be consumed from Scala 3 we need a scala 3 macro implementation as well. Here that second implementation is missing and when we load the class we ignore the Scala 2 macros. This implies that when loaded, StringContext does not have `s`, `f` and `raw` methods.

The usual solution is to add the second macro, but because those are intrinsic macros we can get away by just making them non-macros and let the compiler do its usual transformation of these methods.

Reply:
Can't we interpret `macro ???` like `???` when it doesn't come from scala 2 so that we can keep using the same source?

Reply:
Not under the normal dual macros rules. We would have to add an extra compilation flag for this special case which seems to be overkill in this single case.

Reply:
> which seems to be overkill in this single case.

I disagree, I think that a special case is better than having a copy of file we need to worry about keeping up to date. Furthermore, it would be nice if in scala/scala we didn't just compile the stdlib with dotty but we also ran the stdlib tests, that is only possible if the compiler can handle the normal StringContext correctly if I understand this correctly.

Reply:
I removed the file and special-cased in these macros in `TreeUnpickler`.